Mage.Sets/src/mage/cards/i/IrreverentGremlin.java

```diff
@@ -0,0 +1,54 @@
+package mage.cards.i;
+
+import java.util.UUID;
+
+import mage.MageInt;
+import mage.abilities.common.EntersBattlefieldControlledTriggeredAbility;
+import mage.abilities.costs.common.DiscardCardCost;
+import mage.abilities.effects.common.DoIfCostPaid;
+import mage.abilities.effects.common.DrawCardSourceControllerEffect;
+import mage.abilities.keyword.MenaceAbility;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.CardType;
+import mage.constants.ComparisonType;
+import mage.constants.SubType;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.filter.predicate.mageobject.AnotherPredicate;
+import mage.filter.predicate.mageobject.PowerPredicate;
+
+/**
+ * @author paasar
+ */
+public final class IrreverentGremlin  extends CardImpl {
+    private static final FilterCreaturePermanent creatureWithPower2OrLessFilter = new FilterCreaturePermanent("creature with power 2 or less");
+    static {
+        creatureWithPower2OrLessFilter.add(AnotherPredicate.instance);
+        creatureWithPower2OrLessFilter.add(new PowerPredicate(ComparisonType.FEWER_THAN, 3));
+
+    }
+
+    public IrreverentGremlin(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.CREATURE}, "{1}{R}");
+
+        this.subtype.add(SubType.GREMLIN);
+        this.power = new MageInt(2);
+        this.toughness = new MageInt(2);
+
+        this.addAbility(new MenaceAbility());
+
+        // Whenever another creature you control with power 2 or less enters, you may discard a card. If you do, draw a card. Do this only once each turn.
+        this.addAbility(
+                new EntersBattlefieldControlledTriggeredAbility(
+                        new DoIfCostPaid(
+                                new DrawCardSourceControllerEffect(1),
+                                new DiscardCardCost()),
+                        creatureWithPower2OrLessFilter)
+                        .setTriggersLimitEachTurn(1));
+    }
+
+    private IrreverentGremlin(final IrreverentGremlin card) { super(card); }
```

Comment:
If you use IntelliJ IDEA then press ctrl+alt+L to auto-format new files (xmage uses IntelliJ default code formation style). If it’s another IDE then don’t use.